api/v3/Contact.php

```diff
@@ -466,15 +469,13 @@ function civicrm_api3_contact_delete($params) {
   // restrict permanent delete if a contact has financial trxn associated with it
   $error = NULL;
   if ($skipUndelete && CRM_Financial_BAO_FinancialItem::checkContactPresent([$contactID], $error)) {
-    return civicrm_api3_create_error($error['_qf_default']);
+    throw new API_Exception($error['_qf_default']);
```

Comment:
@eileenmcnaughton i note here your throwing an \API_Exception where your otherwise throwing a \CiviCRM_API3_Exception which one should it be?

Reply:
I was actually  just noticing that - I think API_Exception is right - not for a great logical reason - just  more consistent. The other is thrown by  the wrapper

Reply:
Fixed